### PR TITLE
feat: override config with whitelisted plugins and theme

### PR DESCRIPTION
**Pull request by prestonlimlianjie**
This PR adds a configuration override file for the isomer build process. The main goal of this override file is to only allow whitelisted Jekyll plugins, remote theme, and set the `safe` configuration to true.

The PR partially addresses issues raised in https://github.com/isomerpages/isomer-product/issues/14 and https://github.com/isomerpages/isomer-product/issues/10:
- Enforce the use of either `isomerpages-template@next-gen` as the remote theme in the _config.yml file
- Enforce only the whitelisted Jekyll plugins in the _config.yml file
- Ensure that `safe: true` is set
- Delete the `_plugins` directory
- Check that Gemfile has not been tampered with - originally, we wanted to do `Override the Gemfile with the whitelisted Gemfile`; but Netlify installs the Gemfile before the build command is run, so we would not be able to intercept the Gemfile. Instead, we check that the Gemfile matches our expected Gemfile and throw an error if they do not exactly match

### Tests
**Remote theme**
- [ ] Non-whitelisted remote themes are overridden with `isomerpages-template@next-gen`
- [ ] All Jekyll plugins are overridden with the default values in _config.yml
- [ ] `safe: true` is set in _config.yml
- [ ] Plugins in the `_plugins` directory are ignored
- [ ] The build fails if the Gemfile is not identical to the one in the overrides folder

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -3,7 +3,27 @@
...
+# Delete custom plugins from _plugins folder if any #
+#####################################################
+rm -rf _plugins
+
+#################################################
+# Check that Gemfile has not been tampered with #
+#################################################
+curl https://raw.githubusercontent.com/opengovsg/isomer-build/master/overrides/Gemfile -o Gemfile-template
+diff_line_count_gemfile =$(diff Gemfile Gemfile-template | wc -l)
+if (( diff_line_count_gemfile > 0 )); then
```
Comment (gweiying) [inline build.sh:23]: typo, should be `$diff_line_count_gemfile`

File: build.sh [hunk 2]
```diff
@@ -3,7 +3,27 @@
 #################################################################
 # Override netlify.toml with centrally-hosted netlify.toml file #
 #################################################################
-curl https://raw.githubusercontent.com/opengovsg/isomer-build/master/netlify.toml -o /opt/build/repo/netlify.toml
+curl https://raw.githubusercontent.com/opengovsg/isomer-build/master/overrides/netlify.toml -o /opt/build/repo/netlify.toml
+
+###################################################################
+# Obtain config override file to enforce plugins and remote theme #
+###################################################################
+curl https://raw.githubusercontent.com/opengovsg/isomer-build/master/overrides/_config-override.yml
```
Comment (gweiying) [inline build.sh:11]: should specify download path `/opt/build/repo/_config-override.yml`

Comment (prestonlimlianjie) [reply build.sh]: Fixed in d463b75

File: build.sh (hunk 1 above)
Comment (prestonlimlianjie) [reply build.sh:23]: As discussed, we verified that this line of code works

Comment (gweiying) [reply build.sh:23]: Some small issues in https://app.netlify.com/sites/a-test-v4-staging/deploys/610271407bf6ec000763fd93, see lines `133, 134`

File: build.sh [hunk 3]
```diff
@@ -3,7 +3,27 @@
...
+#####################################################
+# Delete custom plugins from _plugins folder if any #
+#####################################################
+rm -rf _plugins
+
+#################################################
+# Check that Gemfile has not been tampered with #
+#################################################
+curl https://raw.githubusercontent.com/opengovsg/isomer-build/master/overrides/Gemfile -o /opt/build/repo/Gemfile-template
+diff_line_count_gemfile=$(diff /opt/build/repo/Gemfile /opt/build/repo/Gemfile-template | wc -l)
```
Comment (gweiying) [inline build.sh:22]: Should we add the option to ignore whitespace, `diff -b` or `diff -w` https://ss64.com/bash/diff.html

Comment (prestonlimlianjie) [reply build.sh]: added `--ignore-space-change` (equivalent to `-b`) in 64c7396. good suggestion

File: build.sh (hunk 1 above)
Comment (prestonlimlianjie) [reply build.sh:23]: The issue was a typo, fixed in c676a74

**Pull request merged**